### DISCUSSION TITLE
Target Rate Intermittent,  not triggered enough times?

### DISCUSSION THREAD

**Discussion by pabutterworth** [BottlecapDave/HomeAssistant-OctopusEnergy#838] [Q&A]
HI,
I've just started to use the target rate feature to identify the best 3 hours to turn on the immersion heater overnight.

<img width="610" alt="image" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/16760791/7c794434-264d-46a4-9987-84221f3d2a3f">

My setup is as the image above, but it's only triggered 3 times so 1.5 hours, rather than the expected 6 times making 3 hours.

Have I misunderstood the usage?

Thanks
Paul

**Comment by BottlecapDave** [comment]
Hello.

The sensor should have turned on at the correct times. During those 3 times, I assume they were only on for 30 minute chunks?. Please show the history view of the sensor during the effecting time period along with the times selected by the sensor (available in the attritbues).

**Comment by pabutterworth** [comment]
Oh, sorry it's my miskake. I was assuming it would trigger in 30 min slots, if they are consecutive,  it joins them up. Need to re-think my automation.
Thanks

**Comment by solarjones** [comment]
I'm in the same position wanting my immersion to be triggered using myenergi-eddi device and just started using this integration.  I've set my target sensor for intermittent 1.5 hours and when slots are identified but not consecutive, everything seems to work well. Eg. ON event is triggered and I know to turn on the immersion for 30 minutes.  I 'boost' my immersion for 30 minutes for each OE slot identified.But like you, I've found if the 1.5 hours of slots are consecutive, I'm only seeing one ON trigger.   

Did you find a work around? I'm thinking of boosting for 1.5 hours for each ON event, and then force stopping the immersion at the OFF event. Not the best but all I can think of.

@BottlecapDave Is there a way to have the sensor provide an ON event at the start of every slot identified?  

Thanks

**Reply by BottlecapDave** [reply]
The "on" event is supplied by HA and not something I'm in control of. This is the nature of this type of sensor in HA. Every time the sensor turns on you can have a look at the current_duration_in_hours attribute to know how long the sensor will be on for, so you know how long to turn your heater on for.

**Comment by pabutterworth** [comment]
I just have a trigger on ‘on’ and ‘off’. Seems to work a treat.

Paul Butterworth
+44 7768 252250
